src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -3239,13 +3307,25 @@ void AwtWindow::_SetFullScreenExclusiveModeState(void *param)
     jobject self = data->window;
     jboolean state = data->isFSEMState;
 
-    PDATA pData;
-    JNI_CHECK_PEER_GOTO(self, ret);
-    AwtWindow *window = (AwtWindow *)pData;
+    AwtWindow *window = nullptr;
+    if (self == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "self");
+        delete data;
+        return;
+    } else {
+        window = (AwtWindow *) JNI_GET_PDATA(self);
+        if (window == nullptr) {
+            THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
+            env->DeleteGlobalRef(self);
+            delete data;
+            return;
+        }
+    }
 
     window->setFullScreenExclusiveModeState(state != 0);
 
-  ret:
+
```

Comment:
I recommend deleting this line, since it does not make sense to have two consecutive blank lines here.